pqarrow/arrowutils/sort.go

```diff
@@ -0,0 +1,90 @@
+package arrowutils
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/apache/arrow/go/v14/arrow/array"
+	"github.com/apache/arrow/go/v14/arrow/compute"
+	"github.com/apache/arrow/go/v14/arrow/memory"
+)
+
+// SortRecord sorts the given record's rows by the given column. Currently only supports int64, string and binary columns.
+func SortRecord(mem memory.Allocator, r arrow.Record, cols []int) (*array.Int64, error) {
+	if len(cols) > 1 {
+		return nil, fmt.Errorf("sorting by multiple columns isn't implemented yet")
+	}
+	indicesBuilder := array.NewInt64Builder(mem)
+
+	if r.NumRows() == 0 {
+		return indicesBuilder.NewInt64Array(), nil
+	}
+	if r.NumRows() == 1 {
+		indicesBuilder.Append(0)
+		return indicesBuilder.NewInt64Array(), nil
+	}
+
+	indices := make([]int64, r.NumRows())
```

Comment:
I think we have one too many copies than necessary here: one alloc here, then a copy of all indices in `AppendValues` below. I would suggest instead to do:
`indicesBuilder.Reserve(r.NumRows())`
To keep a single allocation then `indicesBuilder.AppendValue` one by one.